pkg/controller/snatglobalinfo.go

```diff
@@ -316,20 +315,11 @@ func (cont *AciController) deleteNodeinfoFromGlInfoCache(nodename string, global
 }
 
 func (cont *AciController) getServiceIps(policy *ContSnatPolicy) (serviceIps []string) {
-	var services []*v1.Service
-	selector := labels.SelectorFromSet(labels.Set(policy.Selector.Labels))
-	cache.ListAll(cont.serviceIndexer, selector,
-		func(servobj interface{}) {
-			services = append(services, servobj.(*v1.Service))
-		})
+        services := cont.getServicesBySelector(labels.SelectorFromSet(
+                            labels.Set(policy.Selector.Labels)),
+                                    policy.Selector.Namespace)
 	for _, service := range services {
-		if len(policy.Selector.Namespace) == 0 || (len(policy.Selector.Namespace) > 0 &&
-			policy.Selector.Namespace == service.ObjectMeta.Namespace) {
-			if len(service.Status.LoadBalancer.Ingress) == 0 {
```

Comment:
This check is required  please put the changes back

Reply:
if len(service.Status.LoadBalancer.Ingress) == 0 {

Reply:
As discussed offline, this check is in the new function to list services